theodoku/budget_app_rails
# RoR Capstone Project - Budget App

In this Pull Request:
**Interactions**

- Splash screen

 [ ] A simple page with the name of app and links to the sign up and log in.

[ ] Home page (categories page)

**When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.**

[ ] Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category 
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.

 [ ] Add a new category page

The user fills out a form to create a new category

 [ ] Add a new expense page

The user fills out a form to create a new transaction with:
name
amount

 [ ] Testing requirements

Create unit and integration tests for all the most important components 

- Use Postgres as your database.

- Use devise for authentication.

- Validate all user input to make sure that anyone with bad intentions cannot compromise  app.
